components/esp_system/Kconfig

```diff
@@ -55,10 +55,10 @@ menu "ESP System Settings"
             wait before the device reboots.
 
     config ESP_SYSTEM_SINGLE_CORE_MODE
-        bool
-        default n
+        bool "Only initialize first core."
+        default "y" if FREERTOS_UNICORE
```

Comment:
I'm afraid there is one issue with this approach. If the project is initially configured with `CONFIG_FREERTOS_UNICORE=y`, this option will get set to `y` as well. However if the developer subsequently changes `CONFIG_FREERTOS_UNICORE` to `n` in menuconfig, this option will still be set to `y`. Kconfig doesn't update the options with already defined values if the default value of the option changes.

Reply:
I think as long as the developer has not changed the default value of CONFIG_ESP_SYSTEM_SINGLE_CORE_MODE, It should change to the default.

However, you do bring up a good point. ESP_SYSTEM_SINGLE_CORE_MODE=y and CONFIG_FREERTOS_UNICORE=n is an invalid configuration. Let me see if i can fix this.

Reply:
I added a 
depends FREERTOS_UNICORE
which should fix the invalid configuration

Reply:
> I think as long as the developer has not changed the default value of CONFIG_ESP_SYSTEM_SINGLE_CORE_MODE, It should change to the default.

Sadly, this is not how Kconfig works! (Would maybe be nice if it did.)

Thanks for the fix.